tests/v2/validation/fleet/snapshot/snapshot_restore_existing_gitrepo_test.go

```diff
@@ -0,0 +1,263 @@
+	"github.com/rancher/shepherd/extensions/workloads"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	namespace      = "fleet-default"
+	containerImage = "nginx"
+)
+
+type FleetWithSnapshotTestSuite struct {
+	suite.Suite
+	client         *rancher.Client
+	session        *session.Session
+	fleetGitRepo   *v1alpha1.GitRepo
+	clustersConfig *etcdsnapshot.Config
+	clusterID      string
+}
+
+func (f *FleetWithSnapshotTestSuite) TearDownSuite() {
+	f.session.Cleanup()
+}
+
+func (f *FleetWithSnapshotTestSuite) SetupSuite() {
+	f.session = session.NewSession()
+
+	client, err := rancher.NewClient("", f.session)
+	require.NoError(f.T(), err)
+
+	f.client = client
+
+	f.fleetGitRepo = &v1alpha1.GitRepo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fleet.FleetMetaName + namegenerator.RandStringLower(5),
+			Namespace: fleet.Namespace,
+		},
+		Spec: v1alpha1.GitRepoSpec{
+			Repo:            fleet.ExampleRepo,
+			Branch:          fleet.BranchName,
+			Paths:           []string{fleet.GitRepoPathLinux},
+			CorrectDrift:    &v1alpha1.CorrectDrift{},
+			ImageScanCommit: v1alpha1.CommitSpec{AuthorName: "", AuthorEmail: ""},
+			Targets:         []v1alpha1.GitTarget{{ClusterName: f.client.RancherConfig.ClusterName}},
+		},
+	}
+
+	f.client, err = f.client.ReLogin()
+	require.NoError(f.T(), err)
+
+	userConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, userConfig)
+
+	clusterObject, _, _ := extensionscluster.GetProvisioningClusterByName(f.client, f.client.RancherConfig.ClusterName, fleet.Namespace)
+	if clusterObject != nil {
+		status := &provv1.ClusterStatus{}
+		err := steveV1.ConvertToK8sType(clusterObject.Status, status)
+		require.NoError(f.T(), err)
+
+		f.clusterID = status.ClusterName
+	} else {
+		f.clusterID, err = extensionscluster.GetClusterIDByName(f.client, f.client.RancherConfig.ClusterName)
+		require.NoError(f.T(), err)
+	}
+
+	podErrors := pods.StatusPods(f.client, f.clusterID)
+	require.Empty(f.T(), podErrors)
+
+	//may not be necessary since this isn't dynamic
+	logrus.Info("loading snapshot config")
+	f.clustersConfig = new(etcdsnapshot.Config)
+	config.LoadConfig(etcdsnapshot.ConfigurationFileKey, f.clustersConfig)
+	logrus.Info("Done")
+}
+
+func (f *FleetWithSnapshotTestSuite) TestSnapshotThenFleetRestore() {
+	snapshotRestoreAll := &etcdsnapshot.Config{
+		UpgradeKubernetesVersion:     "",
+		SnapshotRestore:              "all",
+		ControlPlaneConcurrencyValue: "15%",
+		ControlPlaneUnavailableValue: "3",
+		WorkerConcurrencyValue:       "20%",
+		WorkerUnavailableValue:       "15%",
```

Comment:
I don't think saving these for the tests is what we should do in this case. We don't want people using the same values for this every time. I feel setting a global const will make people think they should be using those instead of setting their own.